pkg/ptrace/ptrace_linux_arm64.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//go:build cgo
+
+package ptrace
+
+import (
+	"encoding/binary"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+var endian = binary.LittleEndian
+
+const syscallInstrSize = 4
+
+const nrProcessVMReadv = 270
+const nrProcessVMWritev = 271
+
+// see kernel source /include/uapi/linux/elf.h
+const nrPRStatus = 1
+
+func getIp(regs *syscall.PtraceRegs) uintptr {
+	return uintptr(regs.Pc)
+}
+
+func getRegs(pid int, regsout *syscall.PtraceRegs) error {
+	return unix.PtraceGetRegSetArm64(pid, nrPRStatus, (*unix.PtraceRegsArm64)(regsout))
+}
+
+func setRegs(pid int, regs *syscall.PtraceRegs) error {
+	return unix.PtraceSetRegSetArm64(pid, nrPRStatus, (*unix.PtraceRegsArm64)(regs))
+}
+
+// Syscall runs a syscall at main thread of process
+func (p *TracedProgram) Syscall(number uint64, args ...uint64) (uint64, error) {
+	// save the original registers and the current instructions
+	err := p.Protect()
+	if err != nil {
+		return 0, err
+	}
+
+	var regs syscall.PtraceRegs
+
+	err = getRegs(p.pid, &regs)
+	if err != nil {
+		return 0, err
```

Comment:
As `getRegs` just invoke `unix.PtraceGetRegSetArm64` directly, I think we need to wrap error here. Same in `ptrace_linux_amd64.go`.